index.js

```diff
@@ -36,13 +78,32 @@ module.exports = function (opts = {}) {
     }
 
     app.authboot = {};
+
+    const concurrency = opts.maxAuthConcurrency
+      || options.maxAuthConcurrency
+      || app.config.get('auth:maxAuthConcurrency')
+      || 3;
+
+    const algorithm = opts.algorithm
+      || options.algorithm
+      || app.config.get('auth:algorithm')
+      || 'sha256';
+
+    let checker;
     const lookup = app.authboot.lookup = (lookupOpt || function ({ name, password }, callback) {
-      const hash = users.get(name);
-      if (!hash) {
+      const hashBuffer = users.get(name);
+      if (!hashBuffer) {
         debug(`unknown username ${name}`);
         return callback(null, false);
       }
-      bcrypt.compare(password, hash, callback);
+      checker = checker || coalesce(concurrency, (password, done) => {
```

Comment:
eslint says to not hide the outer variable `password`

Reply:
running `npm run eslint` does not show this?

Reply:
sorry meant the `eslint in my head` 😆 .  Though eslint _should_ be warning here as it is hiding the outer variable that's called `password`